packages/block-editor/src/components/inserter/media-tab/hooks.js

```diff
@@ -55,22 +55,40 @@ function useInserterMediaCategories() {
 		inserterMediaCategories,
 		allowedMimeTypes,
 		enableOpenverseMediaCategory,
+		registeredInserterMediaCategories,
 	} = useSelect( ( select ) => {
-		const settings = select( blockEditorStore ).getSettings();
+		const { getSettings, getRegisteredInserterMediaCategories } =
+			select( blockEditorStore );
+		const settings = getSettings();
 		return {
 			inserterMediaCategories: settings.inserterMediaCategories,
 			allowedMimeTypes: settings.allowedMimeTypes,
 			enableOpenverseMediaCategory: settings.enableOpenverseMediaCategory,
+			registeredInserterMediaCategories:
+				getRegisteredInserterMediaCategories(),
 		};
 	}, [] );
 	// The allowed `mime_types` can be altered by `upload_mimes` filter and restrict
 	// some of them. In this case we shouldn't add the category to the available media
 	// categories list in the inserter.
 	const allowedCategories = useMemo( () => {
```

Comment:
Don't you think, all this logic in this hook should be a selector. `getInserterMediaCategories`. We might not need the `getRegisteredInserterMediaCategories` selector at all.

Reply:
it could be a private one